src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyTest.java

```diff
@@ -135,4 +135,34 @@ public void discoverNodesOneAddressOnePort()
         assertEquals(new Address(publicAddress, port), node.getPublicAddress());
         assertFalse(iterator.hasNext());
     }
+
+    @Test
+    public void validateValidRegion() {
+        List<String> awsRegions = Arrays.asList(
+            "us-east-2", "us-west-2", "ap-south-1", "ap-northeast-3", "ap-southeast-2",
+            "ca-central-1", "cn-north-1", "cn-northwest-1", "us-gov-east-1");
+
+        awsRegions.forEach(region -> {
+            AwsDiscoveryStrategy awsDiscoveryStrategy = spy(new AwsDiscoveryStrategy(Collections.emptyMap(), null, mockClient));
+            doReturn(region).when(awsDiscoveryStrategy).getCurrentRegion(10, 3);
+
+            awsDiscoveryStrategy.validateRegion(region);
+        });
+    }
+
+    @Test
+    public void validateInvalidRegion() {
+        List<String> invalidRegions = Arrays.asList("us-east-2a", "us-invalid-2","us-gov-1a","us-gov-invalid-2");
+
+        invalidRegions.forEach(region -> {
```

Comment:
I think one invalid region is enough, you don't need to iterate here. If you have some really different scenarios, then create separate test methods.